test/e2e/flask/multichain-api/get-session.spec.ts

```diff
@@ -0,0 +1,146 @@
+import _ from 'lodash';
+import * as path from 'path';
+import {
+  DAPP_URL,
+  largeDelayMs,
+  openDapp,
+  unlockWallet,
+  WINDOW_TITLES,
+  withFixtures,
+} from '../../helpers';
+import { Driver } from '../../webdriver/driver';
+
+import FixtureBuilder from '../../fixture-builder';
+
+const OPTIONS = {
+  dapp: true,
+  dappPaths: [
+    path.join(
+      '..',
+      '..',
+      'node_modules',
+      '@metamask',
+      'test-dapp-multichain',
+      'build',
+    ),
+  ],
+  fixtures: new FixtureBuilder().withPopularNetworks().build(),
+};
+
+async function openDappAndConnectWallet(
+  driver: Driver,
+  extensionId: string,
+): Promise<void> {
+  await unlockWallet(driver);
+
+  await openDapp(driver, undefined, DAPP_URL);
+
+  await driver.fill('[placeholder="Enter extension ID"]', extensionId);
+  await driver.clickElement({ text: 'Connect', tag: 'button' });
+  await driver.delay(largeDelayMs);
+}
+
+describe('Multichain API', function () {
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_getSession` with no session created:', function () {
+    it('should successfully receive empty session scopes', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          ...OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          await openDappAndConnectWallet(driver, extensionId);
+
+          await driver.clickElement('#get-session-btn');
```

Comment:
I know a lot of the e2e tests in the code base use IDs, but Alex and I have been thinking perhaps it's better to target by the text name instead. What are your thoughts?

Reply:
Can you sell me on the upside of targeting by text name instead of button id ?

I thought doing it by ID would be more precise since we should be sure to get the component we want to interact with, where as in getting it by text name could get me a lot of elements with the same text, and I would have to filter the correct one before clicking on it 🤔

Reply:
Resolved in [a105b35](https://github.com/MetaMask/metamask-extension/pull/29298/commits/a105b35b9d82ff0657fb79f9a42d29760b21e09a)